### PR TITLE
Addition of Kokkos sampler (against develop instead of master)

### PR DESCRIPTION
**Pull request by vlkale**
Redoing the PR #140 from @vsurjadidjaja to be against Kokkos Tools develop branch instead of master branch. All PRs of Kokkos Tools should  now (as of a few months ago) go against the develop branch of Kokkos Tools. 

Note that comments of PR #140 and PR #137 still should be addressed.

### CODE REVIEW

**Comment by vlkale** [comment]
Note that the clang-format 8 check is failing at ./common/kokkos-sampler/kp_sampler_skip.cpp 

@vsurjadidjaja

File: profiling/ldms-kokkos-sampler/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+OVIS_DIR=/projects/ovis/streams/opt/ovis
```
Comment (dalg24) [inline profiling/ldms-kokkos-sampler/Makefile:1]: Did you mean to commit that file?

**Comment by dalg24** [comment]
Make sure it can be built from cmake and apply clang-format

**Comment by crtrott** [comment]
Replaced by #169

**Pull request closed without merging**